### PR TITLE
workers: update storage options guide

### PR DESCRIPTION
**Pull request by elithrar**
This PR updates the Storage Options guide (linked to from all storage products) to:

* Better describe ideal use-cases
* Clarify performance / consistency characteristics in context of the guide
* Adds Hyperdrive and Vectorize to the overview

### CODE REVIEW

File: content/workers/platform/storage-options.md [hunk 1]
```diff
@@ -19,17 +19,57 @@ Storage options can also be used by your front-end application built with Cloudf
...
+| ----------------------------- | ------------- | ---------------------- |
+| Key-value storage             | [Workers KV](/workers-kv/) | Configuration data, service routing metadata, personalization (A/B testing) 
+| Object storage                | [R2](/r2/)    | User-facing web assets, images, machine learning and training datasets, analytics datasets, log and event data. |
+| SQL database                  | [D1](/d1/)    | Relational data, including user profiles, product listings and orders, and/or customer data. |
+| Global co-ordination          | [Durable Objects](/durable-objects/) | Building collaborative applications; global co-ordination across clients; strongly consistent, transactional storage. |
+| Vector search (database)      | [Vectorize](/vectorize/) | Storing [embeddings](/workers-ai/models/text-embeddings/) from AI models for semantic search and classification tasks. |
+| Task processing & batching    | [Queues](/queues/) | Background job processing (emails, notifications, APIs) and log processing/batching.
+| Connect to an existing database | [Hyperdrive](/hyperdrive/) | Connecting to an existing database in a cloud or on-prem. |
+
+{{</table-wrap>}}
```
Comment (irvinebroque) [inline content/workers/platform/storage-options.md:46]: 🙌

File: content/workers/platform/storage-options.md [hunk 2]
```diff
@@ -130,23 +142,51 @@ To get started with D1:
...
+* Learn more [about how Queues works](/queues/learning/how-queues-works/).
+
+
+## Hyperdrive
+
+Hyperdrive is a service that accelerates queries you make to existing databases, making it faster to access your data from across the globe, irrespective of your users’ location.
+
+Hyperdrive allows you to:
+
+* Connect to an existing database from Workers without connection overheads.
```
Comment (irvinebroque) [inline content/workers/platform/storage-options.md]:
```suggestion
* Connect to an existing database from Workers without connection overhead.
```

**Review by irvinebroque — APPROVED** [review]
🔥

File: content/workers/platform/storage-options.md [hunk 3]
```diff
@@ -19,17 +19,57 @@ Storage options can also be used by your front-end application built with Cloudf
...
+## Performance and consistency
+
+The following table highlights the performance and consistency characteristics of the primary storage offerings available to Cloudflare Workers:
+
+{{<table-wrap>}}
+
+| Feature                                       | Workers KV   | R2           | Durable Objects      | D1           |
+| --------------------------------------------- | ------------ | ------------ |--------------------- | ------------ |
+| Maximum storage per account                   | Unlimited<sup>1</sup> | Unlimited<sup>2</sup> | 50 GiB            | 50GiB <sup>3</sup><sup>beta</sup>
+| Storage grouping name<sup>2</sup>             | Namespace    | Bucket           | Durable Object       | Database |
```
Comment (vy-ton) [inline content/workers/platform/storage-options.md:58]: Not sure what `Storage grouping name` footnote should reference?

Comment (vy-ton) [inline content/workers/platform/storage-options.md:58]: For DO, can we standardize on `Durable Object Namespace` as the grouping name?

**Review by vy-ton — COMMENTED** [review]
Some things I caught on the live docs

**Pull request merged**